src/modules/fancyzones/lib/FancyZonesData.cpp

```diff
@@ -0,0 +1,609 @@
+#include "pch.h"
+#include "FancyZonesData.h"
+#include "FancyZonesDataTypes.h"
+#include "JsonHelpers.h"
+#include "ZoneSet.h"
+#include "trace.h"
+#include "Settings.h"
+
+#include <common/common.h>
+#include <common/json.h>
+
+#include <shlwapi.h>
+#include <filesystem>
+#include <fstream>
+#include <optional>
+#include <regex>
+#include <sstream>
+#include <unordered_set>
+
+namespace
+{
+    const wchar_t* FANCY_ZONES_DATA_FILE = L"zones-settings.json";
+    const wchar_t* FANCY_ZONES_APP_ZONE_HISTORY_FILE = L"app-zone-history.json";
+    const wchar_t* DEFAULT_GUID = L"{00000000-0000-0000-0000-000000000000}";
+    const wchar_t* REG_SETTINGS = L"Software\\SuperFancyZones";
+
+    const wchar_t ActiveZoneSetsTmpFileName[] = L"FancyZonesActiveZoneSets.json";
+    const wchar_t AppliedZoneSetsTmpFileName[] = L"FancyZonesAppliedZoneSets.json";
+    const wchar_t DeletedCustomZoneSetsTmpFileName[] = L"FancyZonesDeletedCustomZoneSets.json";
+
+    std::wstring ExtractVirtualDesktopId(const std::wstring& deviceId)
+    {
+        // Format: <device-id>_<resolution>_<virtual-desktop-id>
+        return deviceId.substr(deviceId.rfind('_') + 1);
+    }
+
+    const std::wstring& GetTempDirPath()
+    {
+        static std::wstring tmpDirPath;
+        static std::once_flag flag;
+
+        std::call_once(flag, []() {
+            wchar_t buffer[MAX_PATH];
+
+            auto charsWritten = GetTempPath(MAX_PATH, buffer);
+            if (charsWritten > MAX_PATH || (charsWritten == 0))
+            {
+                abort();
+            }
+
+            tmpDirPath = std::wstring{ buffer };
+        });
+
+        return tmpDirPath;
+    }
+}
+
+namespace FancyZonesDataNS
```

Comment:
Is this NS really necessary?

Reply:
Nice catch! Leftover from the early phase :) Fixed!